obspy/core/trace.py

```diff
@@ -2779,6 +3241,207 @@ def _data_sanity_checks(value):
                "arrays are allowed for initialization.") % str(value.shape)
         raise ValueError(msg)
 
+class FrequencyDomainTrace(BaseTrace):
+
+    def ifft(self):
+        tr = Trace()
+        tr.stats = Stats()
+        tr.stats.starttime = self.stats.starttime
+        tr.data = fftpack.irfft(self.data)
```

Comment:
Whatever the reason - it is super confusing and IMHO makes it very hard to work with it. Judging by some scipy issues, the scipy devs seem to largely have the same opinion ;-)

We definitely want data arrays that are contiguous in memory (for eventually talking with any C algorithms) and prepending for example with `np.concatenate()` will create a copy. I'm not sure if there is a way to grow a numpy array without a copy. It might be possible using `realloc()` from C but that is also not guaranteed to not create a copy.